### PR TITLE
fix: add all chart dependency repos, clean up when finished

### PR DESCRIPTION
**Pull request by mrparkers**

### CODE REVIEW

File: src/util/github.js [hunk 1]
```diff
@@ -15,24 +15,24 @@ const getOctokit = () => {
...
 const getRepositoryBranch = (owner, repo, branch) => getOctokit().rest.repos.getBranch({
     owner,
     repo,
-    branch
+    branch,
 });
 
 const getFileFromPages = async (owner, repo, file) => {
-    const { data: { html_url } } = await getOctokit().rest.repos.getPages({
+    const { data } = await getOctokit().rest.repos.getPages({
```
Comment (jknight-liatrio) [inline src/util/github.js:28]: This line looks untested. I haven't run it and I'm not good enough at javascript to know that the return value of data is correct. I'm assuming it's correct, but it looks a little strange

File: src/prepare.js [hunk 2]
```diff
@@ -59,15 +57,15 @@ const prepare = async (
     if (oldChartIndexFile) {
         await fs.writeFile(
             path.join(tempDir, "current-index.yaml"),
-            oldChartIndexFile
+            oldChartIndexFile,
         );
 
-        return await helmRepoIndex(tempDir, chartRepoUrlPrefix, "current-index.yaml");
+        return helmRepoIndex(tempDir, chartRepoUrlPrefix, "current-index.yaml");
```
Comment (jknight-liatrio) [inline src/prepare.js:63]: Where did you move the `await` handler for the `execa` returned by this function?

File: src/util/helm.js [hunk 3]
```diff
@@ -1,23 +1,44 @@
...
+// this is a prerequisite for `helm dependency build`.
+const helmRepoAddDependencies = async (chartPath) => {
+    const chartYamlFile = path.join(chartPath, "Chart.yaml");
+    const chartYaml = await fs.readFile(chartYamlFile);
+
+    const doc = new YAML(chartYaml.toString());
+
+    await Promise.all(doc.json.dependencies.map(async (dependency) => {
+        // the name doesn't matter, in fact it's better that this doesn't possibly conflict with existing helm repos
+        const name = createHash("sha1").update(dependency.repository).digest("hex");
```
Comment (jknight-liatrio) [inline src/util/helm.js]: Do you think there might be any weird side effects in the future of naming helm repo add using a sha hash?

**Review by jknight-liatrio — COMMENTED** [review]
Looks good for the most part there was the one question about the removed `await` on the `helmRepoIndex` call. I could see a potential future issue with naming repos with a sha hash but not a big deal. Test would break and we'd have to fix it later.

File: src/util/github.js (hunk 1 above)
Comment (mrparkers) [reply src/util/github.js:28]: You're right, only the handlers are tested, everything in that `util` folder isn't currently tested.  You can see the coverage results here: https://github.com/liatrio/semantic-release-helm/runs/5761419165?check_suite_focus=true#step:6:10.

But yeah, this change is okay, since I also changed this line: https://github.com/liatrio/semantic-release-helm/pull/11/files#diff-2b9691c756a4997d3facc425fb7e5e4335911b826b273559cc450c66d0a47c1bR28

Basically, we get a `data` object back, which has a property `html_url`.  I just changed how we accessed this property: instead of using a nested [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment), I just accessed it directly.

File: src/prepare.js (hunk 2 above)
Comment (mrparkers) [reply src/prepare.js:63]: This was one of the lint rules that I enabled as a part of this PR.  Since this function is already returning a Promise, the extra `await` here is basically just dead code.

File: src/util/helm.js (hunk 3 above)
Comment (mrparkers) [reply src/util/helm.js]: I can go over my reasoning for this when we chat about the PR

**Review by jknight-liatrio — APPROVED** [review]
